Sources/Chat/ProtocolServices/Invitee/InvitationHandlingService.swift

```diff
@@ -104,46 +108,33 @@ private extension InvitationHandlingService {
 
     enum Errors: Error {
         case inviteForIdNotFound
-        case identityKeyNotFound
-        case inviteKeyNotFound
     }
 
     func setUpRequestHandling() {
-        networkingInteractor.requestSubscription(on: ChatInviteProtocolMethod())
-            .sink { [unowned self] (payload: RequestSubscriptionPayload<InvitePayload>) in
+        networkClient.requestSubscription(on: ChatInviteProtocolMethod())
+            .sink { [unowned self] (payload: RequestSubscriptionPayload<InvitePayload.Wrapper>) in
                 logger.debug("Did receive an invite")
 
-                guard let decoded = try? payload.request.decode()
+                guard let (invite, claims) = try? InvitePayload.decode(from: payload.request)
                 else { fatalError() /* TODO: Handle error */ }
 
                 Task(priority: .high) {
-                    let inviteeAccount = decoded.account
-                    let inviterAccount = try await identityService.resolveIdentity(iss: decoded.iss)
+                    let inviterAccount = try await identityClient.resolveIdentity(iss: claims.iss)
                     // TODO: Should we cache it?
-                    let inviteePublicKey = try await identityService.resolveInvite(account: inviterAccount)
+                    let inviteePublicKey = try await identityClient.resolveInvite(account: inviterAccount)
+                    let inviterPublicKey = invite.inviterPublicKey.did(prefix: false, variant: .X25519)
 
                     let invite = ReceivedInvite(
                         id: payload.id.integer,
-                        message: decoded.message,
+                        message: invite.message,
                         inviterAccount: inviterAccount,
-                        inviteeAccount: inviteeAccount,
-                        inviterPublicKey: decoded.publicKey,
+                        inviteeAccount: invite.inviteeAccount,
+                        inviterPublicKey: inviterPublicKey,
                         inviteePublicKey: inviteePublicKey,
-                        timestamp: decoded.iat // TODO: Replace with relay message receivedAt
+                        timestamp: claims.iat // TODO: Replace with relay message receivedAt
```

Comment:
>TODO: Replace with relay message receivedAt

do we plan to have `receivedAt` does it even make sense?
we know know when we receive message and we know when it was sent with publishedAt